Dockerfile

```diff
@@ -1,78 +1,129 @@
-FROM python:3.7.1-stretch
+# Dockerfile contains multiple levels of images--
+# First, a base image containing dependencies shared by all other images:
+#   - apt dependencies
+#   - rust
+#   - ntl
+#   - pbc
+#   - charm
+#   - base pip dependencies (cython & setup.py)
+# 
+# Thereafter, it adds ever increasing levels of dependencies-- 
+#   - Test requirements (including doc requirements)
+#   - Dev requirements (including aws)
+#
+# In order to build and push this to dockerhub, run:
+# docker build . --target test-image --tag dsluiuc/honeybadgermpc-docker-base
+# docker push dsluiuc/honeybadgermpc-docker-base:latest
+
+FROM dsluiuc/honeybadgermpc-docker-base AS travis-image
```

Comment:
I know you said you will take it up later so just mentioning -> we should not have any reference to Travis here - not even in the name.